### ISSUE TITLE
Need help debugging "Formatting failed"

### ISSUE THREAD

**Issue by Bastes** [mhinz/vim-mix-format#39]
Hi,

Awesome plugin, I've been using it on NeoVim for months now...

...but now for reasons unknown every time I save a file I get a frustrating "Formatting failed" message and no auto formatting.

I don't know where to start looking. I've been checking `:messages` and found nothing out of ordinary. The only tuning I've been doing is a `let g:mix_format_on_save`.

I'm using:
- Ubuntu 20.04.02LTS
- NVIM v0.4.3
- asdf for elixir (1.11.3) and erlang/otp (23)

I've just updated all my nvim plugins just to be on the safe side.

**Comment by Bastes** [comment]
Is there still someone watching over this plugin? :sweat_smile:

**Comment by mhinz** [comment]
Sure, sure. Finding time is another issue, though. :)

Let's try to figure it out...

- I'm assuming `mix` is in $PATH and `mix format` in an Elixir project works as expected.
- Does executing `:MixFormat` manually fail as well?
- 0.4.3 is not the latest release, 0.4.4 is. And even that latest release is quite out-of-date already (by almost 3000 commits). So maybe try a nightly version. E.g. https://github.com/neovim/neovim/releases/download/nightly/nvim-linux64.tar.gz or https://github.com/neovim/neovim/releases/download/nightly/nvim.appimage

**Comment by Bastes** [comment]
Hey :) thanks for taking the time, I didn't mean to be pushy either but I figured if there was a bug it could be interesting to fix it.

So, to answer your questions:
* I'm using fish-shell and installed elixir using `asdf` ; the `~/.asdf/shims` dir is indeed in my `$PATH` and the `mix` shim is in the dir.
* `:MixFormat` fails as well.
* As for the `nvim` nightly, I'll try it to give you more feedback but that'll take a bit of time too ;)

**Comment by Bastes** [comment]
Another thing: is there a way to get more feedback than just `Formatting failed.`? Like a verbose option or something?

**Comment by Bastes** [comment]
Ah, something I just thought to try right now and might be instructive: on another project, formatting succeeds using the vim plugin.

Worth noting is that both project use different elixir versions (`1.11.2` for the one that succeeds, `1.11.3` for the one that fails) but since I did not change the version on the failing project I'm not sure this is the difference.

**Comment by Bastes** [comment]
Aaaand I'm spitballing here but there's also the fact that using `mix format` out of vim succeeds in both projects.

**Comment by Bastes** [comment]
And I just figured it out!

Thank you for rubber ducking for me but apparently the fault was not in the plugin :sweat_smile: it was in one of my configuration file that was making `mix` bug at compile time.

Real sorry to have wasted your time with it. The problem was not manifesting in the dev environment I used to run the command out of vim. And now I'm going to bury myself out of shame :sweat_smile:
